lib/new-admin/graphql/modules/userManagement.js

```diff
@@ -0,0 +1,228 @@
+const bcrypt = require('bcrypt')
+const _ = require('lodash/fp')
+const otplib = require('otplib')
+
+const credentials = require('../../../credentials')
+const loginHelper = require('../../login')
+const users = require('../../../users')
+const sessionManager = require('../../../session-manager')
+const T = require('../../../time')
+
+const REMEMBER_ME_AGE = 90 * T.day
+
+function authenticateUser (username, password) {
+  return loginHelper.checkUser(username).then(hashedPassword => {
+    if (!hashedPassword) return null
+    return Promise.all([bcrypt.compare(password, hashedPassword), hashedPassword])
+  }).then(([isMatch, hashedPassword]) => {
+    if (!isMatch) return null
+    return loginHelper.validateUser(username, hashedPassword)
+  }).then(user => {
+    if (!user) return null
+    return user
+  }).catch(e => {
+    console.error(e)
+  })
```

Comment:
since you're not doing anything in the catch block you can just put catch(console.error) instead